cylc/flow/scheduler.py

```diff
@@ -1046,37 +1056,112 @@ def command_remove_tasks(self, items) -> int:
         """Remove tasks."""
         return self.pool.remove_tasks(items)
 
-    def command_reload_workflow(self) -> None:
+    async def command_reload_workflow(self) -> None:
         """Reload workflow configuration."""
-        LOG.info("Reloading the workflow definition.")
-        old_tasks = set(self.config.get_task_name_list())
-        # Things that can't change on workflow reload:
-        self.workflow_db_mgr.pri_dao.select_workflow_params(
-            self._load_workflow_params
-        )
+        # pause the workflow if not already
+        was_paused = self.is_paused
```

Comment:
This feels like it could be a context manager:

```python
async with pause:
    await self.command_reload_workflow()
```

Reply:
For this usage, we don't always want to unpause the workflow, e.g. if the reload goes wrong. But otherwise, good shout!